### PR TITLE
Remove analyzer dependency support

### PR DESCRIPTION
**Pull request by tmeschter**
We decided that the current design of this feature would not add much to
the user experience, so we're backing out the parts that have been
committed already. Nothing is using these pieces, so this shouldn't even
impact any tests.

### CODE REVIEW

**Comment by tmeschter** [comment]
@srivatsn @ManishJayaswal @mavasani @agocke @jmarolf @shyamnamboodiripad @heejaechang Could you take a look, please?

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 1]
```diff
@@ -56,12 +56,6 @@ public ITaskItem[] Analyzers
             get { return (ITaskItem[])_store["Analyzers"]; }
         }
 
-        public ITaskItem[] AnalyzerDependencies
```
Comment (mavasani) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: Don't you need to update the targets files too: https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L73

**Comment by shyamnamboodiripad** [comment]
:+1:

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs (hunk 1 above)
Comment (tmeschter) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: Goodness, yes.

Comment (tmeschter) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: @mavasani I've fixed this now.

**Comment by mavasani** [comment]
:+1:

**Pull request merged**